### DISCUSSION TITLE
Inference speed - Problems?

### DISCUSSION THREAD

**Discussion by allgrinder** [blakeblackshear/frigate#2540] [Ask A Question] [answered]
Hi, everyone,

Somehow I don't understand the inference speed representations under Debug.

With my normal CPU I reach 10ms. With my coral USB stick, however, only 20ms.
I'm brand new and haven't changed anything, just the basic setting in the yaml.

My setup:
Promox - Home Assistant as VM and there the Frigate addon.
The USB stick is passed on to Hassio.

What am I doing wrong?

Thanks and best regards
![Screenshot_20211230-134048_Home Assistant](https://user-images.githubusercontent.com/75566736/147752823-a9388e1f-3ef9-465c-a13e-208fa6c6fe9a.jpg)

**Comment by blakeblackshear** [comment, marked as answer]
What CPU are you running? Also keep in mind that the inference speed always starts at 10 on startup. It is a rolling average, so make sure it has been running detections before you assume the value is accurate.

**Reply by allgrinder** [reply]
Hey

Thank you for your answer. I have an i5 6400 - a core was used for object recognition. The recommended hardware is shown on the page:
https://docs.frigate.video/hardware - the value is below 15 almost everywhere - so I don't understand my poor value with Coral. What am I doing wrong? Where can I optimize?

Thanks and best regards from Germany

Jan

**Reply by blakeblackshear** [reply]
You want this value to be as low as possible.

**Reply by allgrinder** [reply]
Yes, that's exactly what I understood. But how can my setup only reach 20ms? That is much more powerful than the recommended Nuc.
  So I ask what am I doing wrong?

**Reply by blakeblackshear** [reply]
It's not uncommon to see slower speeds with the USB coral when passed to a VM. That's why the docs specifically recommend running in LXC for proxmox: https://docs.frigate.video/installation#proxmox

**Reply by allgrinder** [reply]
Okay my mistake - I didn't know that;) Thanks for the solution and in advance a happy new year in 2022
